### PR TITLE
[READY] The Cyberbrawlening: Augment your Unarmed Strikes for Maximum Damage

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Robotics can print Advanced cybernetic arms and legs. These limbs have a higher max health (though contribute the exact same amount to core health as normal limbs), higher unarmed damage and higher unarmed effectiveness than organic limbs.

In addition, Strongarm implants now respect the unarmed potential of the limb it occupies. This means you can potentially get much higher damage output if your arm would be capable of higher potential damage and armor penetration. They also do additional stamina damage (1.5x punch damage) so that you can capitalize better on unarmed combinations.

Surplus Prosthetic limbs now contribute a total of 250% of their carried damage against overall health. With only 20 max HP, that means at full damage, the user suffers 50 damage per limb. They also have terrible unarmed values.

Cleans up exofabricators a touch and creates some defines for limb values.

New Sprites 

image

advanced limbs 2

## Why It's Good For The Game

I really want to encourage people to get involved in unarmed combat, so I want to include some potential ways to seek out improving your unarmed combat ability. One possibility should be robotics!

Currently, Strongarm implants are....possibly worse than unarmed attacks? To solve that slightly, they now utilize the attacking limb's unarmed potential as bonuses to the damage dealt as well as penetrative power. I don't have a perfect solution for this at the moment, but at least anyone utilizing this implant isn't unintentionally harming their unarmed combat potency...too much.

Surplus Limbs are, once again, extremely good at taking damage and have almost no actual negatives at the moment. I tried to resolve this in https://github.com/tgstation/tgstation/pull/71739, but the code changes I added have since become obsolete once again, leaving us once again with surplus limbs acting as a health _boost_ rather than a health _malus_. This fixes that issue once and for all, and if you have these limbs, they're going to fucking _suck_.

## Changelog
:cl:
add: The battle against Maint Khan's forces rages on in the periphery stations of the Spinward Sector. And with it, a new breed of unarmed warrior has emerged; the cybernetic martial artist. Nanotrasen, rather than quell the minor maintenance civil war brewing in their sector, have chosen to exploit this conflict to push their weapons and cybernetics research to new heights!
add: Advanced cybernetic arms can be printed at the Robotics exofabricator once researched. They are unlocked by researching the Advanced Robotics Research node. 
add: Advanced cybernetic arms are more durable than standard limbs, and also have higher unarmed potential.
balance: Strongarm implants now utilize the attacking limb's unarmed potential to determine damage and potential armor penetration. It also does additional stamina damage (1.5x punch damage)
balance: Surplus prosthetic limbs contribute more of their carried damage to overall health (AKA they make you actively more vulnerable to damage), and deal less damage with unarmed attacks. Take Quadruple Amputee at your own risk.
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
As I explained on discord, I did. The implant now actively benefits from better limbs and integrates better into unarmed combat.

**Comment by Admiral-Gigaton** [comment]
> What about the strong arm implant? The whole point of this implant was that it made unarmed punches deal more damage and kick people back 1 tile away. Like, why not just change it?

pretty sure they mean strong arm implants when they say "muscle implants" (though there may be a new powergaming roboticist implant i dont know about in that case)

**Comment by excalisword** [comment]
In the same vein as the if already augmented, can I replace my augmented limbs with these better ones, will androids get this better variant of augmented limbs or will they still have the regular version?

**Comment by Raze500** [comment]
Seems interesting!

**Review by Time-Green — APPROVED** [review]
Awesome, I love stuff like this. It seems very overpowered on paper, but considering the trust, resources and time that are involved, I think it's very fair. 

Code looks good, you'll still need a spritetainer approval (when sprites arive)

**Comment by Exester509** [comment]
Little question here, do muscle implants get a buff on normal arms? I haven't really understood how the change in their mechanics affects their performance on standard limbs, though I don't really know much about unarmed combat anyways

File: code/modules/research/techweb/all_nodes.dm [hunk 1]
```diff
@@ -923,10 +923,14 @@
...
 	display_name = "Advanced Robotics Research"
-	description = "Machines using actual neural networks to simulate human lives."
+	description = "Advanced synthetic neural networks and synaptic pathways allows for extraordinary leaps in cybernetic intelligence and interfacing."
 	prereq_ids = list("robotics")
 	design_ids = list(
 		"mmi_posi",
+		"advanced_l_arm",
+		"advanced_r_arm",
+		"advanced_l_leg",
+		"advanced_r_leg",
```
Comment (MrMelbert) [inline code/modules/research/techweb/all_nodes.dm:933]: There's only one other element in this list and you failed the ABC order vibe check, smh

File: code/modules/surgery/bodyparts/robot_bodyparts.dm [hunk 2]
```diff
@@ -514,6 +514,44 @@
...
+// Advanced Limbs: More durable, high punching force
+
+/obj/item/bodypart/arm/left/robot/advanced
+	name = "advanced robotic left arm"
+	desc = "An advanced cybernetic arm, capable of greater feats of strength and durability."
+	unarmed_damage_low = 5
+	unarmed_damage_high = 13
+	unarmed_effectiveness = 20
+	max_damage = 75
+	body_damage_coeff = 0.50
```
Comment (MrMelbert) [inline code/modules/surgery/bodyparts/robot_bodyparts.dm:526]: You should include a comment explaining the whole "more max health, less coeff" thing.

Also this seems easy to break if the default value is changed in the future.

File: code/modules/surgery/organs/internal/cyberimp/augments_arms.dm [hunk 3]
```diff
@@ -436,12 +436,17 @@
 			log_combat(source, target, "attempted to [picked_hit_type]", "muscle implant")
 			return COMPONENT_CANCEL_ATTACK_CHAIN
 
+	var/potential_damage = punch_damage
+	var/obj/item/bodypart/attacking_bodypart = source.get_bodypart(zone)
```
Comment (MrMelbert) [inline code/modules/surgery/organs/internal/cyberimp/augments_arms.dm:440]:
```suggestion
	var/obj/item/bodypart/attacking_bodypart = hand
```
Arm implants already track the arm they're in

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 4]
```diff
@@ -76,6 +76,52 @@
...
 	)
 
+//Advanced Robotic Limbs
+
+/datum/design/advanced_l_arm
+	name = "Advanced Left Arm"
+	id = "advanced_l_arm"
+	build_type = MECHFAB
+	build_path = /obj/item/bodypart/arm/left/robot/advanced
+	materials = list(/datum/material/iron=SHEET_MATERIAL_AMOUNT*10, /datum/material/titanium=SHEET_MATERIAL_AMOUNT*3, /datum/material/gold=SHEET_MATERIAL_AMOUNT*3)
```
Comment (MrMelbert) [inline code/modules/research/designs/mechfabricator_designs.dm:86]:
```suggestion
	materials = list(
		/datum/material/iron = SHEET_MATERIAL_AMOUNT * 10,
		/datum/material/titanium = SHEET_MATERIAL_AMOUNT * 3,
		/datum/material/gold = SHEET_MATERIAL_AMOUNT * 3,
	)
```
I know every other design doesn't multi-line this but it's just so much easier to read at a glance

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 5]
```diff
@@ -76,6 +76,52 @@
...
+	category = list(
+		RND_CATEGORY_CYBERNETICS + RND_SUBCATEGORY_CYBERNETICS_ADVANCED_LIMBS
+	)
+
+/datum/design/advanced_r_arm
+	name = "Advanced Right Arm"
+	id = "advanced_r_arm"
+	build_type = MECHFAB
+	build_path = /obj/item/bodypart/arm/right/robot/advanced
+	materials = list(/datum/material/iron=SHEET_MATERIAL_AMOUNT*10, /datum/material/titanium=SHEET_MATERIAL_AMOUNT*3, /datum/material/gold=SHEET_MATERIAL_AMOUNT*3)
```
Comment (MrMelbert) [inline code/modules/research/designs/mechfabricator_designs.dm:97]:
```suggestion
	materials = list(
	/datum/material/iron = SHEET_MATERIAL_AMOUNT * 10,
	/datum/material/titanium = SHEET_MATERIAL_AMOUNT * 3,
	/datum/material/gold = SHEET_MATERIAL_AMOUNT * 3,
)
```

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 6]
```diff
@@ -76,6 +76,52 @@
...
+	category = list(
+		RND_CATEGORY_CYBERNETICS + RND_SUBCATEGORY_CYBERNETICS_ADVANCED_LIMBS
+	)
+
+/datum/design/advanced_l_leg
+	name = "Advanced Left Leg"
+	id = "advanced_l_leg"
+	build_type = MECHFAB
+	build_path = /obj/item/bodypart/leg/left/robot/advanced
+	materials = list(/datum/material/iron=SHEET_MATERIAL_AMOUNT*10, /datum/material/titanium=SHEET_MATERIAL_AMOUNT*3, /datum/material/gold=SHEET_MATERIAL_AMOUNT*3)
```
Comment (MrMelbert) [inline code/modules/research/designs/mechfabricator_designs.dm:108]:
```suggestion
	materials = list(
	/datum/material/iron = SHEET_MATERIAL_AMOUNT * 10,
	/datum/material/titanium = SHEET_MATERIAL_AMOUNT * 3,
	/datum/material/gold=SHEET_MATERIAL_AMOUNT * 3,
)
```

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 7]
```diff
@@ -76,6 +76,52 @@
...
+	category = list(
+		RND_CATEGORY_CYBERNETICS + RND_SUBCATEGORY_CYBERNETICS_ADVANCED_LIMBS
+	)
+
+/datum/design/advanced_r_leg
+	name = "Advanced Right Leg"
+	id = "advanced_r_leg"
+	build_type = MECHFAB
+	build_path = /obj/item/bodypart/leg/right/robot/advanced
+	materials = list(/datum/material/iron=SHEET_MATERIAL_AMOUNT*10, /datum/material/titanium=SHEET_MATERIAL_AMOUNT*3, /datum/material/gold=SHEET_MATERIAL_AMOUNT*3)
```
Comment (MrMelbert) [inline code/modules/research/designs/mechfabricator_designs.dm:119]:
```suggestion
	materials = list(
		/datum/material/iron = SHEET_MATERIAL_AMOUNT * 10,
		/datum/material/titanium = SHEET_MATERIAL_AMOUNT * 3,
		/datum/material/gold = SHEET_MATERIAL_AMOUNT * 3,
	)
```

File: code/modules/surgery/bodyparts/robot_bodyparts.dm (hunk 2 above)
Comment (necromanceranne) [reply code/modules/surgery/bodyparts/robot_bodyparts.dm:526]: Maybe I should make a define for these + limb baseline, similar to prosthetics? As well as for the coefficient value?

File: code/modules/research/techweb/all_nodes.dm (hunk 1 above)
Comment (necromanceranne) [reply code/modules/research/techweb/all_nodes.dm:933]: I do not think of these things do not bully qwq

File: code/modules/surgery/bodyparts/robot_bodyparts.dm (hunk 2 above)
Comment (MrMelbert) [reply code/modules/surgery/bodyparts/robot_bodyparts.dm:526]: It's prooooobably fine.

**Comment by necromanceranne** [comment]
> In the same vein as the if already augmented, can I replace my augmented limbs with these better ones, will androids get this better variant of augmented limbs or will they still have the regular version?

At the moment, you basically have to amputate, there is no convenient way to upgrade. I'll look into that.

**Comment by DrDiasyl** [comment]
> Little question here, do muscle implants get a buff on normal arms? I haven't really understood how the change in their mechanics affects their performance on standard limbs, though I don't really know much about unarmed combat anyways

Makes you deal 13 brute instead of 5-10. Also gives your punches a knockback (but it can't stun)

**Comment by necromanceranne** [comment]
> Little question here, do muscle implants get a buff on normal arms? I haven't really understood how the change in their mechanics affects their performance on standard limbs, though I don't really know much about unarmed combat anyways

It's a bit of a buff from live. It has the additional unarmed damage, but it also has a broader damage range. On live, it deals a consistent 13 damage. In this pr it deals 10 to 15, with an average of about 13, using standard human limbs. It also has some armor penetration.

**Comment by DrDiasyl** [comment]
@necromanceranne you should make punching with strong-arm-implant/upgraded-cyborg-arm make a unique meaty hit sound.

Also, what you think of allowing to punch windows and structures? No turfs though.

**Comment by necromanceranne** [comment]
I can look into a unique sound, but I feel like it'd be best to leave the window/structure breaking to hulk. Tide implants without the same downsides might be a bit much in my opinion.

File: code/modules/surgery/organs/internal/cyberimp/augments_arms.dm (hunk 3 above)
Comment (necromanceranne) [reply code/modules/surgery/organs/internal/cyberimp/augments_arms.dm:440]: guh

File: code/modules/surgery/bodyparts/robot_bodyparts.dm (hunk 2 above)
Comment (necromanceranne) [reply code/modules/surgery/bodyparts/robot_bodyparts.dm:526]: Let's not take any chances haha

File: code/modules/research/designs/mechfabricator_designs.dm (hunk 4 above)
Comment (necromanceranne) [reply code/modules/research/designs/mechfabricator_designs.dm]: cleaned up the whole thing for ya

**Comment by DATA-xPUNGED** [comment]
> In this pr it deals 10 to 15, with an average of about 13, using standard human limbs. It also has some armor penetration.

the average from 10 to 15 is 12.5, not 13

**Comment by necromanceranne** [comment]
I'm going to key your car.

**Comment by DATA-xPUNGED** [comment]
> I'm going to key your car.

I don't have one

**Comment by necromanceranne** [comment]
Well I lost my key, so we'll call it a stalemate

**Comment by excalisword** [comment]
If I might recommend something, I don't recall many unarmed attacks using legs besides the kick on the ground I believe? I think the advanced robot legs should give a sort of dash maybe?

**Comment by necromanceranne** [comment]
Part of the unarmed combat mechanics is putting people on the ground to get access to the much stronger kicks. This is kind of deliberate. I don't want to overload their benefits too much.

**Comment by excalisword** [comment]
Ah ok, I didn't know kicks were used on people on the ground that much, thought it was more of a flavor thing more than anything.

**Comment by necromanceranne** [comment]
It's a little esoteric I admit, but no it's actually mechanically advantageous to kick.

**Comment by excalisword** [comment]
So is it randomly determined whether you kick or punch or is it just that you kick someone automatically when they are on the ground.

**Comment by deathrobotpunch** [comment]
> > In the same vein as the if already augmented, can I replace my augmented limbs with these better ones, will androids get this better variant of augmented limbs or will they still have the regular version?
> 
> At the moment, you basically have to amputate, there is no convenient way to upgrade. I'll look into that.



> So is it randomly determined whether you kick or punch or is it just that you kick someone automatically when they are on the ground.

you kick someone instead of punching when they are on the ground. and on a related note did you know that kicking someone with mag boots that are on they deal 30 damage?

**Comment by excalisword** [comment]
> > So is it randomly determined whether you kick or punch or is it just that you kick someone automatically when they are on the ground.
> 
> you kick someone instead of punching when they are on the ground. and on a related note did you know that kicking someone with mag boots that are on they deal 30 damage?

This is funny, but if its only limited to on magboots then its fine, and its funny to do your best Issac Clarke impression.

**Comment by Loafin34** [comment]
what about adding brass knuckles, stuff that you can put in your glove slot to augment like strongarm? no knockback of course, but it could make easier wounds.

**Comment by nukularpower** [comment]
Love this!  hopefully this means that hulk can be un-nerfed a bit too :D

**Comment by deathrobotpunch** [comment]
> what about adding brass knuckles, stuff that you can put in your glove slot to augment like strongarm? no knockback of course, but it could make easier wounds.

man that would be so cool, finally the greytide can punch holes in the heads of others.

**Comment by deathrobotpunch** [comment]
also I noticed with the implants is that they increase damage directly,

what I would have preferred was a punch force buff which would make the damage increase but vary each punch and make it easier to inflict wounds. 

this way it would actually give an increased sense of strength to the augmented person and allows unarmed combat to be more exciting instead of the old ‘click until down’ unarmed combat system we have right now

**Comment by carlarctg** [comment]
I use surplus limbs and they are miserable to FIX, not just have generally. It's super annoying to heal robolimbs. If you're gonna make them conventionally terrible look into that as well

**Comment by F-e-r-n** [comment]
I cant wait to do a cyberpsycho gimmick as a sleeping carp tator

**Comment by DAKKA-WAAAGH** [comment]
We cyberpogging, lads

**Comment by deathrobotpunch** [comment]
> I cant wait to do a cyberpsycho gimmick as a sleeping carp tator

damn that would be so OP, but I have no problem with that due to the research and surgery needed. but the thing is, does this work with the auto surgeon?

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 8]
```diff
@@ -76,6 +76,68 @@
...
+	name = "Advanced Left Arm"
+	id = "advanced_l_arm"
+	build_type = MECHFAB
+	build_path = /obj/item/bodypart/arm/left/robot/advanced
+	materials = list(
+		/datum/material/iron=SHEET_MATERIAL_AMOUNT*10,
+		/datum/material/titanium=SHEET_MATERIAL_AMOUNT*3,
+		/datum/material/gold=SHEET_MATERIAL_AMOUNT*3,
+	)
+	construction_time = 200
```
Comment (Ghommie) [inline code/modules/research/designs/mechfabricator_designs.dm:91]:
```suggestion
	construction_time = 20 SECONDS
```
Also goes for the other designs you're adding.

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Pull request merged**